hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/key/OMKeyRenameResponse.java

```diff
@@ -93,6 +93,15 @@ public void addToDBBatch(OMMetadataManager omMetadataManager,
       omMetadataManager.getKeyTable().putWithBatch(batchOperation,
```

Comment:
Not sure, if we need to do this. 
- In case of replay its only deleting fromKey, in this case we do not have the toKeyName available, its NULL,
- In the first tie, it seems the sequence would be : 
    - update mapping id <-> toKeyname
    - add toKey
    - delete FromKey

So in case of replay if the newkey is already there then we should also have the id-<->keyname mapping updated.

I also noticed that the sequence of delete and newkey addition was reversed before this change. Not sure how this worked.

Reply:
Let me elaborate on the scenario I am talking about:
  Trxn n1 : Create Key - key1
  Trxn n2 : Commit Key - key1 (adds n1 -> key1 entry to KeyID table)
  Trxn n3 : Rename key1 -> key2 (update n1 -> key2 in KeyID table)

Now lets say we replay transactions from n1. 
  Replay Trxn n1 : Create Key - key1 (key1 does not exist so this transaction is replayed)
  Replay Trxn n2 : Commit Key - key1 (this again updates n1 -> key1 in KeyID table)
  Replay Trxn n3 : Since key2 exists, we figure out this is replay and execute 
                              deleteFromKeyOnly() logic in addToBatch.

So finally after replay, we have n1 in KeyID table pointing to key1 instead of key2.

> I also noticed that the sequence of delete and newkey addition was reversed before this change. Not sure how this worked.

The order does not matter here as they are part of batch operation.

Reply:
Thanks @prashantpogde for the offline discussion. 
It will be lot of extra work updating the keyID in deleteFromKeyOnly case also as we don't send the toKey name in this case. It is safe to go ahead with this approach as Bharat's replay optimization - HDDS-3354 will take care of this.